libsmtutil/Z3Interface.h

```diff
@@ -50,7 +50,7 @@ class Z3Interface: public SolverInterface, public boost::noncopyable
 	// This is used to make the runs more deterministic and platform/machine independent.
 	// The tests start failing for Z3 with less than 20000000,
 	// so using double that.
```

Comment:
Lie ;-P!

Reply:
Apart from not claiming that we use double of it, we could also check, if it's still actually the case that tests start failing with lower values at all (but we can defer that).

Reply:
> Lie ;-P!

Is it still a lie if there's no intent? ;)

Reply:
You're really bothered by that, aren't you :-D?

Reply:
I just want you to admit that a `lie` does **not** require intent.

Reply:
Well :-D... as far as I'm concerned the actor who's doing the "lying" here would be the comment itself - it doesn't make sense to say the original author lied, because at the time of writing it was true - and whoever changes the code rendering the comment incorrect doesn't exactly lie either, if they don't even touch the comment.
But still any reader of the comment in the future will be given incorrect information, i.e. be lied to :-).
But since a comment in a source file traditionally doesn't have a mind of its own and thereby one can't really consider it to have intent, I grant you that the presence of a ``lie`` doesn't imply the presence of intent. Happy :-)?

Reply:
In any case: just change the comment ;-P!

Reply:
Happy, seems to be an uncommon view these days.
Will change, I'm binary searching the break point, but won't try much further

Reply:
12500000 works
10000000 fails the current tests
I would keep 20000000 with the reasoning we had before "double the first that fails"

Reply:
Wikipedia says "A lie is an assertion that is believed to be false, typically used with the purpose of deceiving someone.".
And ``Typically != Generally`` and ``Typically != Necessarily`` :-).

Reply:
Wikipedia is oversimplifying the subject, it can go much deeper: https://plato.stanford.edu/entries/lying-definition/